src/commands/toolchain.rs

```diff
@@ -64,12 +86,27 @@ fn name_allowed(s: &str) -> Result<String> {
     }
 }
 
+fn channel_allowed(s: &str) -> Result<String> {
+    if s.is_empty() {
+        return Ok(s.to_string());
+    }
+    if toolchain_override::Channel::from_str(s).is_err() {
+        bail!(
+            "Invalid channel '{}', expected one of {}.",
+            s,
+            VALID_CHANNEL_STR,
+        );
+    }
+    Ok(s.to_string())
```

Comment:
Could we return a string slice too?

Reply:
Why should this be returned a string slice, do you means that we need to support for export with multi channels?

Reply:
Little less work for the compiler if we don't turn it into a String. No biggie